kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/marketplace/view/MarketplaceFragment.java

```diff
@@ -0,0 +1,173 @@
+
+		//Spend Recycler
+		RecyclerView spendRecycler = root.findViewById(R.id.spend_recycler);
+		spendRecycler.setLayoutManager(new LinearLayoutManager(getActivity(), LinearLayoutManager.HORIZONTAL, false));
+		spendRecycler.addItemDecoration(itemDecoration);
+		spendRecyclerAdapter = new SpendRecyclerAdapter();
+		spendRecyclerAdapter.bindToRecyclerView(spendRecycler);
+		spendRecyclerAdapter.setOnItemClickListener(new OnItemClickListener() {
+			@Override
+			public void onItemClick(BaseRecyclerAdapter adapter, View view, int position) {
+				marketplacePresenter.onItemClicked(position, OfferType.SPEND);
+			}
+		});
+		spendRecyclerAdapter.setEmptyView(new OffersEmptyView(getContext()));
+
+		//Earn Recycler
+		RecyclerView earnRecycler = root.findViewById(R.id.earn_recycler);
+		earnRecycler.setLayoutManager(new LinearLayoutManager(getActivity(), LinearLayoutManager.HORIZONTAL, false));
+		earnRecycler.addItemDecoration(itemDecoration);
+		earnRecyclerAdapter = new EarnRecyclerAdapter();
+		earnRecyclerAdapter.bindToRecyclerView(earnRecycler);
+		earnRecyclerAdapter.setOnItemClickListener(new OnItemClickListener() {
+			@Override
+			public void onItemClick(BaseRecyclerAdapter adapter, View view, int position) {
+				marketplacePresenter.onItemClicked(position, OfferType.EARN);
+			}
+		});
+		earnRecyclerAdapter.setEmptyView(new OffersEmptyView(getContext()));
+
+	}
+
+
+	@Override
+	public void setSpendList(List<Offer> spendList) {
+		spendRecyclerAdapter.setNewData(spendList);
+	}
+
+	@Override
+	public void setEarnList(List<Offer> earnList) {
+		earnRecyclerAdapter.setNewData(earnList);
+	}
+
+	@Override
+	public void showOfferActivity(PollBundle pollBundle) {
+		try {
+			Intent intent = PollWebViewActivity.createIntent(getContext(), pollBundle);
+			startActivity(intent);
+			getActivity()
+				.overridePendingTransition(R.anim.kinecosystem_slide_in_right, R.anim.kinecosystem_slide_out_left);
+		} catch (ClientException e) {
+			marketplacePresenter.showOfferActivityFailed();
+		}
+
+	}
+
+	@Override
+	public void showSpendDialog(ISpendDialogPresenter spendDialogPresenter) {
+		SpendDialog spendDialog = new SpendDialog(getActivity(), marketplacePresenter.getNavigator(),
+			spendDialogPresenter);
+		spendDialog.show();
+	}
+
+	@Override
+	public void showToast(String msg) {
+		Toast.makeText(getContext(), msg, Toast.LENGTH_SHORT).show();
+	}
+
+	@Override
+	public void notifyEarnItemRemoved(int index) {
+		earnRecyclerAdapter.itemRemoved(index);
+	}
+
+	@Override
+	public void notifyEarnItemInserted(int index) {
+		earnRecyclerAdapter.itemInserted(index);
+	}
+
+	@Override
+	public void notifySpendItemRemoved(int index) {
+		spendRecyclerAdapter.itemRemoved(index);
+	}
+
+	@Override
+	public void notifySpendItemInserted(int index) {
+		spendRecyclerAdapter.itemInserted(index);
+	}
+
+	@Override
+	public void showSomethingWentWrong() {
+		showToast(getString(R.string.kinecosystem_something_went_wrong));
```

Comment:
need to protect UI thread? in public methods that affect UI (and should be run on UI thread?)